boringNotch/components/Notch/NotchHomeView.swift

```diff
@@ -131,6 +145,118 @@ struct NotchHomeView: View {
 //                    .opacity(vm.notchState == .closed ? 0 : 1)
 //                    .blur(radius: vm.notchState == .closed ? 20 : 0)
             }
+            .onAppear {
+                // Initialize the slider value and start the timer
+                sliderValue = musicManager.elapsedTime
+                startTimer()
+            }
+            .onDisappear {
+                // Stop the timer when the view disappears
+                timer?.cancel()
+            }
+        }
+    }
+
+    private func startTimer() {
+        timer = Timer.publish(every: 0.1, on: .main, in: .common)
+            .autoconnect()
+            .sink { [self] _ in
+                self.updateSliderValue()
+            }
+    }
+
+    private func updateSliderValue() {
+        guard !dragging else { return } // Only update if not dragging
+        guard musicManager.isPlaying else { return }
+        let currentTime = Date()
+        let timeDifference = currentTime.timeIntervalSince(musicManager.timestampDate)
+        // Calculate the real-time elapsed time
+        let currentElapsedTime = musicManager.elapsedTime + (timeDifference * musicManager.playbackRate)
+        sliderValue = min(currentElapsedTime, musicManager.songDuration) // Clamp to song duration
+    }
+}
+
+struct MusicSliderView: View {
+    @Binding var sliderValue: Double
+    var duration: Double
```

Comment:
Make the duration to be `@State`

Reply:
Should this be `@Binding`?(idk I don't have a lot of swift experience)

Reply:
You are right, `@Binding`